qprofiler2/src/org/qcmg/qprofiler2/summarise/ReadGroupSummary.java

```diff
@@ -205,354 +251,291 @@ else if(! PairedRecordUtils.isCanonical(  record) ){
+		//add to xml RG_Counts
+		Element ele =  XmlUtils.createMetricsNode(parent, "pairs", pairNum.get());
 
-		return new Pair<Integer, String>(modal.getLeft() , String.format( "%,.2f",  stdDev.getResult()));
+		XmlUtils.outputValueNode( ele, "mateUnmappedPair", mateUnmapped.get() );
+		XmlUtils.outputValueNode( ele, "mateDifferentReferencePair", diffRef.get() );
 		
-	}
-	
-	
-//	@Deprecated
-//	public void iSize2Xml(Element parent) { 
-//		Element rgElement = QprofilerXmlUtils.createSubElement(parent, node_readgroup);
-//		rgElement.setAttribute("id", readGroupId );
-//		final int boundary = Math.min(farTlenValue *2, bigTlenValue/2);
-//
-//				
-//		Pair<Integer, String> isize = getIsizeStats();
-//		rgElement.setAttribute( modal_isize, isize.getLeft() + "" );
-//		rgElement.setAttribute("stdDev", isize.getRight());
-//				
-//		//counts for each Tlen value
-//		int start = 0;
-//		for(int i = 0; i < boundary ; i ++ ) 
-//			if(isize1.get(i) != 0) 			
-//				setISizeElement(rgElement,isize1.get(i),   i, i);				
-//				
-//		start = boundary;
-//		long count = isize1.get(boundary); 
-//		for(int i = boundary ; i < bigTlenValue;  i ++ ){
-//			count += isize1.get(i);
-//			if( i % 1000 == 0 ){
-//				if(count != 0)
-//					setISizeElement(rgElement, count,   start, i);
-//				count = 0; 
-//				start = i+1; 				
-//			}			
-//		}
-//		
-//		//region for last bin before bigTlenValue
-//		if(count > 0)
-//			setISizeElement(rgElement, count,   start, (bigTlenValue - 1));
-//				
-//		//region for oversize Tlen 
-//		setISizeElement(rgElement, isize1.get(bigTlenValue),   bigTlenValue, max_isize.get() );
-//	}
-	
-	/**
-	 * Create element node for each isize region with same counts
-	 * @param parent
-	 * @param count
-	 * @param start
-	 * @param end
-	 */
-	private void setISizeElement(Element parent, long count, int start, int end ){
-		Document doc = parent.getOwnerDocument(); 
-		Element cycleE = doc.createElement("RangeTallyItem");
-		cycleE.setAttribute("count", count +"");
-		cycleE.setAttribute("start", start + "");
-		cycleE.setAttribute("end", end + "");	
-		parent.appendChild(cycleE);			
-	}
+		f5f3.toXml( ele );
+		f3f5.toXml( ele );
+		inward.toXml( ele );
+		outward.toXml( ele );		
 
-	private double addDiscardReadStats(Element parent, String nodeName, AtomicLong counts, long totalReads ){
-			//add to RG_count section
-			Element stats = QprofilerXmlUtils.createSubElement(parent, nodeName);	
-			stats.setAttribute("readCount", String.format("%d",counts.get()));
-						 	
-			//duplicats, non canonical and unmapped will be counted, others reads just discards
-			float percentage = 100 * (float) counts.get() / totalReads ;						
-			stats.setAttribute("basePercent",  String.format("%2.2f", percentage   ));		
-			
-			return percentage; 
-			 	 
+		//output tlen refer to method lostBaseStats(...)
+		long sum = 0, no = 0;
+		for(int i = 1; i < middleTlenValue; i ++ ){
+			sum += isize.get(i) * i;
```

Comment:
perhaps call get once, store the value and use again?

Reply:
what do you mean?

Reply:
apologies - i thought isize was a map - calling get on a map twice for the same key is what I was trying to avoid